pkg/registry/core/pod/storage/eviction.go

```diff
@@ -324,13 +333,22 @@ func addConditionAndDeletePod(r *EvictionREST, ctx context.Context, name string,
 			return podObj, nil
 		}
 
-		podCopyUpdated := rest.DefaultUpdatedObjectInfo(pod, conditionAppender)
+		podCopyUpdated := rest.DefaultUpdatedObjectInfo(podCopy, conditionAppender)
 
-		if _, _, err = r.store.Update(ctx, name, podCopyUpdated, rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{}); err != nil {
+		updatedPodObject, _, err := r.store.Update(ctx, name, podCopyUpdated, rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
+		if err != nil {
 			return err
 		}
+		if !resourceVersionIsUnset(deleteOptionsCopy) {
+			newResourceVersion, err := meta.NewAccessor().ResourceVersion(updatedPodObject)
+			if err != nil {
+				return err
+			}
+			// bump the resource version, since we are the one who modified it via the update
+			deleteOptionsCopy.Preconditions.ResourceVersion = &newResourceVersion
+		}
 	}
-	_, _, err := r.store.Delete(ctx, name, rest.ValidateAllObjectFunc, options)
+	_, _, err = r.store.Delete(ctx, name, rest.ValidateAllObjectFunc, deleteOptionsCopy)
```

Comment:
one think that could be problematic is if somebody does update after our update, before we try to delete the pod. In that case we should do a rollback of the condition. Do we care about such a race? Should I implement the rollback? @liggitt

Reply:
I don't know... I thought we had a controller reconciling bogus conditions

Reply:
We have a disruption controller which cleans up stale DisruptionTarget conditions: https://github.com/kubernetes/kubernetes/blob/152876a3eb9c6211c48bb159c87e2bf834e4eedc/pkg/controller/disruption/disruption.go#L729. By stale we mean conditions which are 2min old if there was no deletion in the meanwhile.

Reply:
Cool, I think, because it is an unlikely race, it should be enough to leave it to the controller